### PR TITLE
DashR callbacks chapter updates

### LINKED ISSUE #833: Dash R Callbacks chapter missing subchapters
The Dash for R docs currently lack the Dash Callbacks chapter which encompasses 4 other subchapters on "advanced" callback features.  To achieve parity with both sets of documentation,  the Dash for R docs should be updated to reflect the new chapters and the content within them to include clientside callbacks, advanced callbacks, and other hidden information currently in the FAQ.

On the Python side, these changes were added in #813 .

### PR DESCRIPTION
**Pull request by HammadTheOne**
This PR addresses changes to the Callbacks chapters and subchapters in Dash for R docs. The PR also includes some minor fixes to examples, chapter numbering, and some restructuring the match the Python docs. The PR adds chapters for advanced callbacks, clientside callbacks, a "gotchas" section, and updates to the multiple outputs documentation.  

Closes #833 

Post-merge checklist:

The master branch is auto-deployed to `dash.plotly.com`.
Once you have merged your PR, wait 5-10 minutes and check dash.plotly.com
to verify that your changes have been made.

- [ ] I understand

If this PR documents a new feature of Dash:

- [ ] Comment on the original Dash issue with a link to the new docs.
- [ ] Reply to any community thread(s) asking for this feature.

If this PR includes a new dataset available at a remote URL:
- [ ] I have added this dataset to the `datasets/` folder
- [ ] I have added a mapping between the remote URL and the filename in the
`datasets/` folder into the `find_and_replace` dict in `dash_docs/tools.py`

If this PR adds an image or animated GIF:
- [ ] This image was saved and referenced locally rather than via an external link

If I introduced a new relative link inside `dcc.Markdown`:
- [ ] I considered whether I could replace the `dcc.Markdown` call with `reusable_components.Markdown`, which will replace relative links with `tools.relpath` internally. Otherwise, I used e.g. `<dccLink href=tools.relpath('/layout') children="the first chapter"/>` instead of `[the first chapter](/layout)` (importing `tools` from `dash_docs`), and set `dangerously_allow_html=true` in the `dcc.Markdown` call.

If I changed the `chapter_index` by removing or relocating a page:
- [ ] I added a redirect in `dash_docs/server.py` from the old URL to the new URL

### CODE REVIEW

File: dash_docs/chapters/advanced_callbacks/index.R [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+library(dashCoreComponents)
+library(dashHtmlComponents)
+
+utils <- new.env()
+source('dash_docs/utils.R', local=utils)
+
+examples <- list(
+  prevent.update.button=utils$LoadExampleCode('dash_docs/chapters/advanced_callbacks/examples/prevent_update_button.R'),
+  prevent.update=utils$LoadExampleCode('dash_docs/chapters/advanced_callbacks/examples/prevent_update.R'),
+  last_clicked_button=utils$LoadExampleCode('dash_docs/chapters/advanced_callbacks/examples/last_clicked_button.R')
```
Comment (chriddyp) [inline dash_docs/chapters/advanced_callbacks/index.R:11]: Let's standardize on a naming convention that matches the filename. So, `last_clicked_button` is 👍  but let's change `prevent.update.button` to `prevent_update_button`.

That'll make it easier eventually to create a function that generates this list automatically like the new `tools.load_examples` function available in Python.

**Comment by chriddyp** [comment]
Overall looks good, thank you for making this change off of the `nights-since-september` branch! 

One thing that is missing is the redirects that were added in https://github.com/plotly/dash-docs/pull/827 on the python side.

File: dash_docs/chapters/advanced_callbacks/index.R (hunk 1 above)
Comment (HammadTheOne) [reply dash_docs/chapters/advanced_callbacks/index.R:11]: Thanks for pointing it out, fixed in f47e72b.

**Comment by rpkyle** [comment]
> Overall looks good, thank you for making this change off of the `nights-since-september` branch!
> 
> One thing that is missing is the redirects that were added in #827 on the python side.

added in d58a643

**Pull request merged**